test/src/SpendPermissions/getHash.t.sol

```diff
@@ -0,0 +1,16 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.23;
+
+import {Test, console2} from "forge-std/Test.sol";
+
+import {SpendPermissions} from "../../../src/SpendPermissions.sol";
+
+import {SpendPermissionsBase} from "../../base/SpendPermissionsBase.sol";
+
+contract GetHashTest is Test, SpendPermissionsBase {
+    function setUp() public {
+        _initializeSpendPermissions();
+    }
+
+    function test_getHash_success() public {}
```

Comment:
Would add cases to check for replay conditions. Does the same recurring allowance have the same hash if we switch contracts or chains?